facebook/junitTests/src/test/java/com/facebook/share/internal/ShareContentValidationTest.java

```diff
@@ -0,0 +1,313 @@
+    public void testItDoesNotAcceptShareVideoContentByWeb() {
+        SharePhoto previewPhoto = buildSharePhoto("https://facebook.com/awesome.gif");
+        ShareVideoContent shareVideoContent =
+                new ShareVideoContent.Builder().setPreviewPhoto(previewPhoto).build();
+
+        ShareContentValidation.validateForWebShare(shareVideoContent);
+    }
+
+    // Share by Api
+    @Test(expected = FacebookException.class)
+    public void testItValidatesNullContentForApiShare() {
+        ShareContentValidation.validateForApiShare(null);
+    }
+
+    @Test(expected = FacebookException.class)
+    public void testItValidatesNullImageForSharePhotoContentByApi() throws MalformedURLException {
+        SharePhotoContent.Builder spcBuilder = new SharePhotoContent.Builder();
+        SharePhoto sharePhoto = new SharePhoto.Builder().setImageUrl(null)
+                .build();
+        SharePhotoContent sharePhotoContent = spcBuilder.addPhoto(sharePhoto).build();
+
+        ShareContentValidation.validateForApiShare(sharePhotoContent);
+    }
+
+    // Valid Share Contents
+    @Test
+    public void testItAcceptNullImageForShareLinkContent() throws MalformedURLException, URISyntaxException {
+        ShareLinkContent nullImageContent = buildShareLinkContent(null);
+
+        ShareContentValidation.validateForApiShare(nullImageContent);
+        ShareContentValidation.validateForMessage(nullImageContent);
+        ShareContentValidation.validateForNativeShare(nullImageContent);
+        ShareContentValidation.validateForWebShare(nullImageContent);
+    }
+
+    @Test
+    public void testItAcceptsHttpForShareLinkContent() throws MalformedURLException, URISyntaxException {
+        Uri imageUri = Uri.parse("http://facebook.com/awesome-content.gif");
+        ShareLinkContent linkContent = buildShareLinkContent(imageUri);
+
+        ShareContentValidation.validateForApiShare(linkContent);
+        ShareContentValidation.validateForMessage(linkContent);
+        ShareContentValidation.validateForNativeShare(linkContent);
+        ShareContentValidation.validateForWebShare(linkContent);
+    }
+
+    @Test
+    public void testItAcceptsHttpsForShareLinkContent() throws MalformedURLException, URISyntaxException {
+        Uri imageUri = Uri.parse("https://facebook.com/awesome-content.gif");
+        ShareLinkContent linkContent = buildShareLinkContent(imageUri);
+
+        ShareContentValidation.validateForApiShare(linkContent);
+        ShareContentValidation.validateForMessage(linkContent);
+        ShareContentValidation.validateForNativeShare(linkContent);
+        ShareContentValidation.validateForWebShare(linkContent);
+    }
+
+    @Test
+    public void testItAcceptsSharePhotoContent() throws MalformedURLException, URISyntaxException {
+        String AppId = "200";
+
+        PowerMockito.mockStatic(Validate.class);
+        when(Validate.hasContentProvider(RuntimeEnvironment.application)).thenReturn(true);
```

Comment:
Instead of the return true and changing the other classes you can do something along the lines of 
doNothing().when(Validate.class, "hasContentProvider", RuntimeEnvironment.application);